src/pybind/mgr/restful/api.py

```diff
@@ -0,0 +1,659 @@
+            else:
+                commands.append({
+                    'prefix': 'osd down',
+                    'ids': [str(osd_id)]
+                })
+
+        if 'reweight' in args:
+            commands.append({
+                'prefix': 'osd reweight',
+                'id': osd_id,
+                'weight': args['reweight']
+            })
+
+        return module.instance.submit_request([commands])
+
+
+
+class OsdIdCommand(Resource):
+    _endpoint = '/osd/<int:osd_id>/command'
+
+    @catch
+    @auth
+    def get(self, osd_id):
+        """
+        Show implemented commands for the OSD ID
+        """
+        osd = module.instance.get_osd_by_id(osd_id)
+
+        if not osd:
+            return {'message': 'Failed to identify the OSD id "%d"' % osd_id}, 500
+
+        if osd['up']:
+            return common.OSD_IMPLEMENTED_COMMANDS
+        else:
+            return []
+
+
+
+class OsdIdCommandId(Resource):
+    _endpoint = '/osd/<int:osd_id>/command/<string:command>'
+
+    @catch
+    @auth
+    def post(self, osd_id, command):
+        """
+        Run the implemented command for the OSD id
+        """
+        osd = module.instance.get_osd_by_id(osd_id)
+
+        if not osd:
+            return {'message': 'Failed to identify the OSD id "%d"' % osd_id}, 500
+
+        if not osd['up'] or command not in common.OSD_IMPLEMENTED_COMMANDS:
+            return {'message': 'Command "%s" not available' % command}, 500
+
+        return module.instance.submit_request([[{
+            'prefix': 'osd ' + command,
+            'who': str(osd_id)
+        }]])
+
+
+
+class Pool(Resource):
+    _endpoint = '/pool'
+
+    @catch
+    @auth
+    def get(self):
+        """
+        Show the information for all the pools
+        """
+        pools = module.instance.get('osd_map')['pools']
+
+        # pgp_num is called pg_placement_num, deal with that
+        for pool in pools:
+            if 'pg_placement_num' in pool:
+                pool['pgp_num'] = pool.pop('pg_placement_num')
+
+        return pools
+
+
+    @catch
+    @auth
+    def post(self):
+        """
+        Create a new pool
+        Requires name and pg_num dict arguments
+        """
+        args = json.loads(request.data)
+
+        # Check for the required arguments
+        pool_name = args.pop('name', None)
+        if pool_name is None:
+            return {'message': 'You need to specify the pool "name" argument'}, 500
+
+        pg_num = args.pop('pg_num', None)
+        if pg_num is None:
+            return {'message': 'You need to specify the "pg_num" argument'}, 500
+
+        # Run the pool create command first
+        create_command = {
+            'prefix': 'osd pool create',
+            'pool': pool_name,
+            'pg_num': pg_num
+        }
+
+        # Check for invalid pool args
+        invalid = common.invalid_pool_args(args)
+        if invalid:
+            return {'message': 'Invalid arguments found: "%s"' % str(invalid)}, 500
```

Comment:
I was thinking about that but for the sake of consistency I used 500 almost everywhere (apart from auth where I used 401 to prompt for login/password). I suppose the important thing here is it is not 2xx.